### PR TITLE
[Docs] Added dartpad demo for animation controller

### PR DESCRIPTION
**Pull request by Alabhya268**
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

https://user-images.githubusercontent.com/57143358/108379104-9e4def80-722b-11eb-8fcd-3cede54c25f9.mp4

### CODE REVIEW

**Comment by Alabhya268** [comment]
@shihaohong @Piinks, Please review this PR.

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 1]
```diff
@@ -195,6 +195,82 @@ enum AnimationBehavior {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=freeform}
+///
```
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animation_controller.dart:199]: There should be an explanation here telling the reader what this sample code shows.

Comment (goderbauer) [reply packages/flutter/lib/src/animation/animation_controller.dart:199]: Also: There's already a lot of example code on this class. What is this showing that's not already covered by the other examples?

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 2]
```diff
@@ -195,6 +195,82 @@ enum AnimationBehavior {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=freeform}
```
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animation_controller.dart:198]: This code can probably be simplified by using stateful_widget_material_ticker as template?

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 3]
```diff
@@ -195,6 +195,82 @@ enum AnimationBehavior {
...
+/// class AnimationControllerDemo extends StatefulWidget {
+///   @override
+///   _AnimationControllerDemoState createState() =>
+///       _AnimationControllerDemoState();
+/// }
+///
+/// class _AnimationControllerDemoState extends State<AnimationControllerDemo>
+///     with SingleTickerProviderStateMixin {
+///   static const Duration _duration = Duration(seconds: 1);
+///   dynamic controller;
```
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animation_controller.dart:218]: Fields should be properly typed.

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 4]
```diff
@@ -195,6 +195,82 @@ enum AnimationBehavior {
...
+///
+///   @override
+///   void initState() {
+///     super.initState();
+///     controller = AnimationController(vsync: this, duration: _duration)
+///       ..addListener(() {
+///         setState(() {});
+///       });
```
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animation_controller.dart:226]: An empty setState call usually isn't idiomatic. You'd probably want to use an AnimatedBuilder to only rebuild the part of the tree that depends on the controller value.

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 5]
```diff
@@ -195,6 +195,82 @@ enum AnimationBehavior {
...
+///           ),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+///
 /// The assumption in the code above is that the animation controllers are being
```
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animation_controller.dart:274]: This block of text belongs to the snippet above the sample code you added.

**Review by goderbauer — COMMENTED** [review]
I am not sure how useful this example really is. What is it trying to show people?

**Comment by Alabhya268** [comment]
> I am not sure how useful this example really is. What is it trying to show people?

As for me personally I had to see animation controller working code to understand it, So I thought it would be nice to have dart pad demo in docs for this.

File: packages/flutter/lib/src/animation/animation_controller.dart (hunk 2 above)
Comment (Alabhya268) [reply packages/flutter/lib/src/animation/animation_controller.dart:198]: ok

File: packages/flutter/lib/src/animation/animation_controller.dart (hunk 3 above)
Comment (Alabhya268) [reply packages/flutter/lib/src/animation/animation_controller.dart:218]: I made it dynamic to fix null safety , As analyse linux test was failing.

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 6]
```diff
@@ -195,6 +195,82 @@ enum AnimationBehavior {
...
+///               ConstrainedBox(
+///                 constraints: BoxConstraints(maxWidth: 200),
+///                 child: Text(
+///                   '${controller.value.toStringAsFixed(2)}',
+///                   style: Theme.of(context).textTheme.headline3,
+///                   textScaleFactor: 1.0 + controller.value,
+///                 ),
+///               ),
+///               ElevatedButton(
+///                 child: Text('animate'),
```
Comment (Abhishek01039) [inline packages/flutter/lib/src/animation/animation_controller.dart:255]: you should add `const` before the Text widget.

**Comment by Alabhya268** [comment]
@goderbauer , I've now changed the template and removed unnecessary code. Please review this PR.

**Comment by Alabhya268** [comment]
@goderbauer , I now changed the template and added description to this demo, Please review this PR again.

**Comment by Alabhya268** [comment]
@goderbauer , I now changed the template and added description to this demo, Please review this PR again.

**Comment by Alabhya268** [comment]
@shihaohong please review this PR.

File: packages/flutter/lib/src/animation/animation_controller.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/lib/src/animation/animation_controller.dart:205]: Is it possible to convert the existing samples here into dartpads?

**Comment by Alabhya268** [comment]
@Piinks @shihaohong Please review this PR.

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 7]
```diff
@@ -203,6 +203,70 @@ enum AnimationBehavior {
 /// controllers are created in [State.initState] and disposed in
 /// [State.dispose], as described in the previous section.)
 ///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
```
Comment (goderbauer) [inline packages/flutter/lib/src/animation/animation_controller.dart:206]: As noted before, there's already a lot of example code on this class. It's not clear to me what this one is supposed to show that's not already clear from the other examples.

Comment (Alabhya268) [reply packages/flutter/lib/src/animation/animation_controller.dart:206]: Ok I will be closing this PR.

**Pull request closed without merging**